async-http-client-backend/fs2/src/main/scala/sttp/client3/asynchttpclient/fs2/AsyncHttpClientFs2Backend.scala

```diff
@@ -49,8 +51,8 @@ class AsyncHttpClientFs2Backend[F[_]: Async] private (
       override def publisherToBytes(p: Publisher[ByteBuffer]): F[Array[Byte]] = {
         p.toStream[F]
           .compile
-          .fold(ByteBuffer.allocate(0))(concatByteBuffers)
-          .map(_.array())
+          .fold(new ConcurrentLinkedQueue[Array[Byte]]())(enqueueBytes)
```

Comment:
do we need a `ConcurrentQueue` here? Wouldn't an ordinary `List`/`Vector`/`Queue` suffice? The fold is sequential, adding elements one-by-one, providing the previous value

Reply:
right, I'll replace it with normal linked queue

Reply:
or just a Scala immutable Queue? :)

Reply:
yes